lib/python/script/db.py

```diff
@@ -231,3 +231,52 @@ def db_commit_transaction(driver):
     if driver in ('sqlite', 'pg', 'mysql'):
         return 'COMMIT'
     return ''
+
+
+_SQL_INT_TYPES = [
+    "INT",
+    "INTEGER",
+    "TINYINT",
+    "SMALLINT",
+    "MEDIUMINT",
+    "BIGINT",
+    "UNSIGNED BIG INT",
+    "INT2",
+    "INT8",
+]
+
+
+_SQL_FLOAT_TYPES = [
```

Comment:
What about NUMERIC and DECIMAL?

Reply:
The [db_sqltype_to_Ctype()](https://grass.osgeo.org/programming7/sqlCtype_8c_source.html#l00024) function returns `DB_C_TYPE_STRING` for NUMERIC and DECIMAL. That's implicit there. They are handled explicitly at other places in the code.

The [Affinity Name Examples](https://sqlite.org/datatype3.html#affinity_name_examples) for SQLite has NUMERIC and DECIMAL(a,b) under (its own) NUMERIC together with BOOLEAN and DATETIME, not under REAL. NUMERIC in SQLite seems different than in [PostgreSQL](https://www.postgresql.org/docs/13/datatype-numeric.html) where NUMERIC has the same definition as DECIMAL. (I don't know if or how that's handled in the GRASS db drivers.)

Conclusion: I don't know what to make up of that, but perhaps text representation is a better fit for DECIMAL floating point number (1) (you are using DECIMAL after all). So, I would leave it as is.

Note 1: That seems to be what SQLite driver is doing in create_table.c and fetch.c, however PostgreSQL and MySQL drivers in create_table.c change DECIMAL and NUMERIC to DOUBLE.